src/plugins/external_tool_support/src/bowtie/BowtieSupport.cpp

```diff
@@ -65,4 +66,16 @@ BowtieSupport::BowtieSupport(const QString& id)
     toolKitName = "Bowtie";
 }
 
+QString BowtieSupport::checkPaths(const QStringList& arguments) const {
+    QStringList errors;
+    if (isOsWindows()) {
+        errors.append(ExternalToolSupportUtils::checkArgumentPathSymbols(arguments));
```

Comment:
Do not write a copy-paste code like this.
Make it in a declarative way: add flags to every tool that will force the common base code check for certain props.